launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -108,6 +113,23 @@ static boolean isWindows() {
     return os.startsWith("Windows");
   }
 
+  /**  Returns an enum value indicating whose JVM is being used. */
+  static JavaVendor getJavaVendor() {
+    String vendorString = System.getProperty("java.vendor");
+    if (vendorString.contains("Oracle")) {
+      return JavaVendor.Oracle;
+    } else {
```

Comment:
Minor: This doesn't have to have the `else`s and thus extra level or two of nesting.